drivers/serial/uart_stm32.c

```diff
@@ -1147,20 +1147,14 @@ static int uart_stm32_async_init(const struct device *dev)
 
 	data->uart_dev = dev;
 
-	if (data->dma_rx.dma_name != NULL) {
-		data->dev_dma_rx = device_get_binding(data->dma_rx.dma_name);
-		if (data->dev_dma_rx == NULL) {
-			LOG_ERR("%s device not found", data->dma_rx.dma_name);
-			return -ENODEV;
-		}
+	if (data->dma_rx.dma_dev != NULL) {
+		__ASSERT_NO_MSG(device_is_ready(data->dma_rx.dma_dev));
```

Comment:
The test of uart async api not working. When I made this change:
```
if(!device_is_ready(data->dma_rx.dma_dev)) {
	return -ENODEV;
}
```
It worked

Reply:
Strange, this was tested ok. I'll have a new look.
Thanks for reporting.

Reply:
@shlomow can you check status using https://github.com/zephyrproject-rtos/zephyr/pull/32716 ?

Reply:
@shlomow, can  you check status using https://github.com/zephyrproject-rtos/zephyr/pull/32716?

Reply:
Fixed, txs